### ISSUE TITLE
What do you think about using MQTTX?

### ISSUE THREAD

**Issue by ysfscream** [emqx/MQTTX#771]
### Discussed in https://github.com/emqx/MQTTX/discussions/770

<div type='discussions-op-text'>

<sup>Originally posted by **ysfscream** November  8, 2021</sup>
First of all, thank you for using MQTT X. We would like to discuss here some comments about using MQTT X. 

We will collect and display some of your good comments on the [ official website](https://mqttx.app) so that more people can learn about MQTTX, and for the bad aspects, we will carefully organize them into task tracking and improve them in time. We'll also show your GitHub account or company link and name.

You can reply according to the following template, which can be directly copied and pasted.

```
#### Github or Company: xxx

#### 👍Good

#### 👎Bad
```

**Comment by zhanghuanchong** [comment]
I will choose 👍🏽  if the #766 can be fixed. It looks like it's easy to fix, but can be helpful to improve the user experience.

**Comment by ysfscream** [comment]
@zhanghuanchong Thanks. In fact, it has been fixed, but not yet released. This issue will be fixed in 1.7.0 to be released within this week.

**Comment by wahmedswl** [comment]
Its really awesome app but needs enhancements. It should really have place to keep snippets! something akin to collection and ability to run that collection.

**Comment by ysfscream** [comment]
> Its really awesome app but needs enhancements. It should really have place to keep snippets! something akin to collection and ability to run that collection.

Hi, Thanks for the suggestion, cloud you please give us some concrete examples of this idea? We are currently saving snippets of payloads or topics.

**Comment by wahmedswl** [comment]
Hi, i don't see anything in the ui to same snippets etc? Its something akin to Postman, it has collection feature with variables etc as well which can be changed per environment etc.
Usually when testing, we only tweak DeviceId in the events and need a place to keep MQTT payloads. I guess, there is really need for Postman for MQTT.

![image](https://user-images.githubusercontent.com/15020561/168310919-2f96efaa-486c-444b-8d04-d77bab130ecc.png)

**Comment by ysfscream** [comment]
@wahmedswl In fact, the main thing is that I didn't expose this operation for the snippet, a payload is automatically saved after each successful send, you just need to click the toggle button to find some history of the sent.

<img width="458" alt="image" src="https://user-images.githubusercontent.com/21299158/168506277-f6c4e9ee-41f6-44d9-ba29-b647c46f35f6.png">

**Comment by wahmedswl** [comment]
Well, the screenshot i show isn't history by the way but this helps :-)
Just need a place to keep payloads, so that we can replay them and with variables to run them in multiple environment. You must have used Postman so something akin to that.

**Comment by ysfscream** [comment]
Yes, MQTT X actually refers to the Postman for many things, Postman is very respectable, but we can't progress that fast because of some resource issues, we can only optimize a little bit, but in the future with sufficient resources, I will still go further and optimize many shortcomings.

But so, many Thanks. Maybe you could join in too, it's a fully open-source community and project. Lol :)

**Comment by wahmedswl** [comment]
Yup, there is no push! MQTT.fx, got snabbed and its too much costly and MQTTX is really capable both in term of ui/features and performance.

Thanks

**Comment by hofq** [comment]
On Windows the App seem to Run in the Background after closing, without a tray icon. I don't like that.

**Comment by ysfscream** [comment]
> On Windows the App seem to Run in the Background after closing, without a tray icon. I don't like that.

I haven't tested it in detail on windows, meaning is it still running in the background after exiting?

**Comment by hofq** [comment]
> > On Windows the App seem to Run in the Background after closing, without a tray icon. I don't like that.
> 
> I haven't tested it in detail on windows, meaning is it still running in the background after exiting?

Yes

**Comment by ysfscream** [comment]
> > > On Windows the App seem to Run in the Background after closing, without a tray icon. I don't like that.
> > 
> > 
> > I haven't tested it in detail on windows, meaning is it still running in the background after exiting?
> 
> Yes

I created a new issue to track this issue. https://github.com/emqx/MQTTX/issues/1020

**Comment by hofq** [comment]
> > > > On Windows the App seem to Run in the Background after closing, without a tray icon. I don't like that.
> > > 
> > > 
> > > I haven't tested it in detail on windows, meaning is it still running in the background after exiting?
> > 
> > Yes
> 
> I created a new issue to track this issue. https://github.com/emqx/MQTTX/issues/1020

Oh, sorry. I have just read the Title of the Issue, not the Description.


I use MQTTX, because it makes managing and communicating with MQTT a lot easier and gives you a great overview on all features of the Protocol

**Comment by ysfscream** [comment]
@hofq Thanks! We fix the issue on Windows as soon as possible.

**Comment by ULTRAKID** [comment]
Any plans to add feature of connection duplication? I have to export json, edit and then import to do this.

**Comment by ysfscream** [comment]
> Any plans to add feature of connection duplication? I have to export json, edit and then import to do this.

Hi! We have supported it in the latest version. https://github.com/emqx/MQTTX/releases/tag/v1.8.2

<img width="396" alt="image" src="https://user-images.githubusercontent.com/21299158/183359441-00716d41-556b-4657-8f25-06fa24d45ef2.png">

**Comment by ULTRAKID** [comment]
> 

Got it! But I don't see the feature in release note😀 I will try it！
![image](https://user-images.githubusercontent.com/28863299/183360184-76fa203d-ee11-45f9-a4c7-cd2313fee7b3.png)

**Comment by ysfscream** [comment]
@ULTRAKID YES. Because we already support it in 1.8.0 😊 Thanks for your feedback and use.

**Comment by rct** [comment]
My initial try of MQTTX v1.9.2 on Windows 11 was very underwhelming. 
* It seemed like it hung/was unresponsive.  It kept running in the background after I closed it.  When I tried to restart it, it wouldn't do anything because it was still hung on whatever in the background. I had to go find it in task manager and kill it.
* A number of things are counterintuitive.  There are no tooltips if you hover over any of the icon buttons.
* It doesn't make very good/intuitive use of screen real estate.  The most prominent icon button in the top left corner takes you to the web site.  Once you click the info button (bottom left), it isn't clear how to get rid of that panel and get back to the main display.  Why isn't that part of help about instead of taking up space for a more commonly used function?
* I couldn't figure out how to do anything useful with the received messages.

**Comment by ysfscream** [comment]
Hi, @rct 
Thank you very much for your suggestions and replies:

1. There were some issues with version 1.9.2. We will release 1.9.3 as soon as possible. Please re-download the Windows version from [mqttx.app](http://mqttx.app/).
2. 3. I fully agree with your views. Currently, space utilization needs to be more reasonable and intuitive.
4. Finally, I would like to know the purpose of handling received messages. What approaches do you expect?

Thank you again for your patience and support. We will keep working hard to enhance the experience with MQTTX. Please feel free to contact us if you have any questions or concerns.

**Comment by brodrigz** [comment]
UI/UX could use some small improvements, the tooltips on the topic list stops you from selecting other topics, wish you could resize the list container horizontally as well, otherwise 10/10 client, really glad its open source

**Comment by ysfscream** [comment]
@brodrigz Thank you so much! We will try our best to optimize according to your suggestion if you provide more details.
